src/Factories/MigrationFactory.php

```diff
@@ -10,11 +10,23 @@ class MigrationFactory
 {
     public function makeFromFile(MigrationFile $file): MigrationInterface
     {
-        require_once $file->path();
+        $class = require_once $file->path();
 
-        $className = Str::studly(implode('_', array_slice(explode('_', $file->name()), 4)));
-        /** @var MigrationInterface $migration */
-        $migration = resolve($className);
+        if (is_object($class)) {
+            // This is an annonymous class so return it directly
+            return $class;
+        }
+
+        try {
+            $className = Str::studly(implode('_', array_slice(explode('_', $file->name()), 4)));
+            /** @var MigrationInterface $migration */
+            $migration = resolve($className);
+        } catch (\Illuminate\Contracts\Container\BindingResolutionException) {
```

Comment:
Do you think this is needed? File names should be unique, how can we get in a situation when `require_once` is called twice?